### ISSUE TITLE
Internal Server Error while creating foreign Keys

### ISSUE THREAD

**Issue by chiragsanghvi** [informatics-isi-edu/ermrest#77]
@karlcz Basically when you create new tables and try creating a foreign key immediately after creation, you get a `500 Internal Server error`.

### Request

```javascript
POST /ermrest/catalog/71/schema/EUVAv/table/dataset_preview/foreignkey HTTP/1.1
HOST: dev.isrd.isi.edu
cookie: ermrest=C6KFIQn2JS37CGovofWnjKfu; 
content-type: application/json
content-length: 219
 
 {"foreign_key_columns":[{"table_name":"dataset_preview","schema_name":"EUVAv","column_name":"preview_id"}],"annotations":{},"referenced_columns":[{"table_name":"sample_name","schema_name":"EUVAv","column_name":"term"}]}
```

### Response

```html
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>500 Internal Server Error</title>
</head><body>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error or
misconfiguration and was unable to complete
your request.</p>
<p>Please contact the server administrator at 
 root@localhost to inform them of the time this error occurred,
 and the actions you performed just before this error.</p>
<p>More information about this error may be available
in the server error log.</p>
</body></html>
```

After 10-15 seconds, if you try creating the foreign key again, the call succeeds.

**Comment by karlcz** [comment]
This is fixed in master so you'll see it once it rolls out to the CI dev server

**Comment by karlcz** [comment]
@robes  see fix in db1de8b4f36d960be21045196fbd0bcae814294c and test cases in dd13799fc070b5878d98e398e8ed75fb78c652da
